core/src/data-generator/custom-dataset.ts

```diff
@@ -0,0 +1,228 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+
+import { PreBundledPysparkJobExecutable } from '../common/pre-bundled-pyspark-job-executable';
+import { CloudWatchEncryptionMode, GlueVersion, JobBookmarksEncryptionMode, PythonVersion, S3EncryptionMode, SecurityConfiguration, WorkerType } from '@aws-cdk/aws-glue-alpha';
+import { PreparedDataset } from './prepared-dataset';
+import { Construct } from 'constructs';
+import { ManagedPolicy, PolicyDocument, PolicyStatement, Role, ServicePrincipal } from 'aws-cdk-lib/aws-iam';
+import { Bucket, Location } from 'aws-cdk-lib/aws-s3';
+import { AraBucket } from '../ara-bucket';
+import { SynchronousGlueJob } from '../synchronous-glue-job';
+import { StringParameter } from 'aws-cdk-lib/aws-ssm';
+import { Aws, Duration, RemovalPolicy } from 'aws-cdk-lib';
+import { AwsCustomResource, AwsCustomResourcePolicy, PhysicalResourceId } from 'aws-cdk-lib/custom-resources';
+import { Key } from 'aws-cdk-lib/aws-kms';
+
+
+/**
+ * The properties for the Bring Your Own Data generator
+ */
+export interface CustomDatasetProps {
+  /**
+   * The S3 location of the input data
+   */
+  readonly s3Location: Location;
+  /**
+   * The format of the input data
+   */
+  readonly inputFormat: CustomDatasetInputFormat;
+  /**
+   * The datetime column to use for data generation as the time reference
+   */
+  readonly datetimeColumn: string;
+  /**
+   * The datetime columns to use for data generation
+   */
+  readonly datetimeColumnsToAdjust: string[];
+  /**
+   * The interval to partition data and optimize the data generation in Minutes
+   */
+  readonly partitionRange: Duration;
+  /**
+   * Approximate data size (in GB) of the custom dataset. 
+   * @default - The Glue job responsible for preparing the data uses autoscaling with a maximum of 100 workers
+   */
+  readonly approximateDataSize?: number;
+}
+
+export enum CustomDatasetInputFormat {
+  CSV = 'csv',
+  PARQUET = 'parquet',
+  JSON = 'json',
+}
+
+/**
+ * A CustomDataset is a dataset that you need to prepare for the [BatchReplayer](@link BatchReplayer) to generate data.
+ * The dataset is transformed into a [PreparedDataset](@link PreparedDataset) by a Glue Job that runs synchronously during the CDK deploy.
+ * The Glue job is sized based on the approximate size of the input data or uses autoscaling (max 100) if no data size is provided.
+ * 
+ * The Glue job is applying the following transformations to the input dataset:
+ * 1. Read the input dataset based on its format. Currently, it supports data in CSV, JSON and Parquet
```

Comment:
Can we provide more clarity on the date format as this is something essential for transforming and replaying the data set?

Reply:
added in construct doc